test/enginetest.c

```diff
@@ -121,8 +121,10 @@ static int test_engines(void)
     display_engine_list();
 
     /*
-     * Depending on whether there's any hardware support compiled in, this
-     * remove may be destined to fail.
+     * At this point, only |new_h1| should remain in the list of engines,
+     * so this removal should empty the list of available engines.
+     * If not, some engine got loaded by other means that our additions
+     * above, and adding |new_h1| again below may fail...
```

Comment:
This comment still makes no sense to me. At this point |new_h1| should *not* be in the list of engines at all (we tried to remove it). Only if some other engine was loaded by some other means would |new_h1| still be in the list

Reply:
Ah, you're right.  It just so happens that when the config file was loaded in my trial run, it also loaded the dyn engine, which ended up first in the list, so that removal at line 81 ended up not removing `new_h1`, but the dyn engine instead, so I had `new_h1` as the lone engine in the list at this point.

Will fix the comment, again.

Reply:
Rewrote the comment.  I hope it's satisfactory this time.